code/modules/power/smes.dm

```diff
@@ -158,9 +158,44 @@
 			to_chat(user, "<span class='alert'>You need more wires.</span>")
 			return
 
-		//build the terminal and link it to the network
-		make_terminal(user)
-		terminal.connect_to_network()
+		if(user.loc == loc)
+			to_chat(user, "<span class='warning'>You must not be on the same tile as the [src].</span>")
+			return
+
+		//Direction the terminal will face to
+		var/tempDir = get_dir(user, src)
+		switch(tempDir)
+			if(NORTHEAST, SOUTHEAST)
+				tempDir = EAST
+			if(NORTHWEST, SOUTHWEST)
+				tempDir = WEST
+		var/turf/tempLoc = get_step(src, reverse_direction(tempDir))
+		if(istype(tempLoc, /turf/space))
+			to_chat(user, "<span class='warning'>You can't build a terminal on space.</span>")
+			return 
+		else if(istype(tempLoc))
+			if(tempLoc.intact)
+				to_chat(user, "<span class='warning'>You must remove the floor plating first.</span>")
+				return 
+		
+		to_chat(user, "<span class='notice'>You start adding cable to the [src].</span>")
+		playsound(src.loc, C.usesound, 50, 1)
+
+		if(do_after(user, 50, target = src))
+			if(!terminal && panel_open)
+				T = get_turf(user)
+				var/obj/structure/cable/N = T.get_cable_node() //get the connecting node cable, if there's one
+				if(prob(50) && electrocute_mob(usr, N, N, 1, TRUE)) //animate the electrocution if uncautious and unlucky
+					do_sparks(5, 1, src)
+					return
+
+				C.use(10) // make sure the cable gets used up

```

Comment:
Might be just me being paranoid, but is there a check at any point to ensure that the cable indeed have 10 cable or more?

Reply:
lines 157-159
```
if(C.amount < 10)
			to_chat(user, "<span class='alert'>You need more wires.</span>")
			return
```

Reply:
Yup, what Shadow pasted are the lines for that